### PR TITLE
Feature/sign up

### PR DESCRIPTION
**Pull request by kevin-rodriguez**
## Links:
* Added sign up feature
* Added session persistence with localStorage
* Added log out feature

### Session persistence and log out
https://user-images.githubusercontent.com/30328809/128925339-0bbe937f-54ad-4270-b31a-8ebb626b76a8.mov

### Sign up
https://user-images.githubusercontent.com/30328809/128925248-622a8fb3-a62f-4d5f-9b8c-e80d58065122.mov

## Risks, Mitigation & Rollback:
- [√] Safe to Revert

### CODE REVIEW

File: template/src/components/common/Button/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import './styles.css';
+
+const Button = ({ children, type = 'button', disabled, handleClick }) => (
+  <button className="button" onClick={handleClick} type={type} disabled={disabled}>
+    {children}
+  </button>
+);
+
+Button.propTypes = {
+  handleClick: func,
```
Comment (nsantos16) [inline template/src/components/common/Button/index.js:12]: I think this prop is required

Comment (nsantos16) [reply template/src/components/common/Button/index.js:12]: Also we need to declare `children` on the propTypes as a `node.isRequired`

File: template/src/components/common/Button/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  disabled: bool,
+  type: string,
```
Comment (nsantos16) [inline template/src/components/common/Button/index.js:14]: I'm wondering if there is something to validate button types in the same way Typescript does 🤔

File: template/src/constants/constants.js [hunk 3]
```diff
@@ -3,3 +3,6 @@ export const DEFAULT_LANGUAGE = 'en';
 
 // REGEX
 export const PASSWORD_REGEX = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
+
+// AUTH
+export const AUTH_USER_KEY = 'user';
```
Comment (fsarachu) [inline template/src/constants/constants.js:8]: probably a good a idea to later read this key from an env variable or something similar

File: template/src/pages/Home/index.js [hunk 4]
```diff
@@ -1,16 +1,26 @@
...
 
 import logo from 'assets/logo.svg';
 
 import './styles.css';
 
 const Home = () => {
   const t = useTranslation();
+  const [logout, { isLoading }] = useLogoutMutation();
+
+  const handleLogout = () => logout().then(() => localStorage.removeItem('user'));
```
Comment (fsarachu) [inline template/src/pages/Home/index.js:13]: Feels like `localStorage.removeItem('user')` would better be placed into the service. 
Always a good idea to keep important logic outside UI components.

File: template/src/pages/form/Login/index.js [hunk 5]
```diff
@@ -34,20 +38,21 @@ const Login = () => {
...
-  const handleFocus = () => {
-    if (error) {
-      resetErrors();
-    }
-  };
+  const handleFocus = () => error && resetErrors();
 
   useEffect(() => {
     if (isSuccess) {
+      setLoggedInUser(user);
```
Comment (fsarachu) [inline template/src/pages/form/Login/index.js:45]: Same here, we should find a way of moving this line to the `useAuth` hook or some similar place

File: template/src/utils/auth.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { AUTH_USER_KEY } from 'constants/constants';
+
+export const setLoggedInUser = user => localStorage.setItem(AUTH_USER_KEY, JSON.stringify(user));
+
+export const getLoggedInUser = () => localStorage.getItem(AUTH_USER_KEY);
```
Comment (fsarachu) [inline template/src/utils/auth.js:5]: How about doing the `JSON.parse` here?

**Review by fsarachu — APPROVED** [review]
Excellent work! 👏 
Left a couple of non-blockers, we can merge as it is and address those in another PR.

**Review by natokra — APPROVED** [review]

**Comment by kevin-rodriguez** [comment]
Merging this as it is and @acastelli will take care of fixing the comments in a new PR

**Pull request merged**